tests/codegen/i128-x86-align.rs

```diff
@@ -0,0 +1,103 @@
+// only-x86_64
+// compile-flags: -O -C no-prepopulate-passes --crate-type=lib
+
+// On LLVM 17 and earlier LLVM's own data layout specifies that i128 has 8 byte alignment,
+// while rustc wants it to have 16 byte alignment. This test checks that we handle this
+// correctly.
+
+// CHECK: %ScalarPair = type { i32, [3 x i32], i128 }
+// CHECK: %Struct = type { i32, i32, [2 x i32], i128 }
+
+#![feature(core_intrinsics)]
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct ScalarPair {
+    a: i32,
+    b: i128,
+}
+
+#[no_mangle]
+pub fn load(x: &ScalarPair) -> ScalarPair {
```

Comment:
Is it intentional that this is testing the rust abi? The C ABI returns ScalarPair quite differently using an sret pointer. https://github.com/rust-lang/rust/pull/130899 will change the Rust ABI to also return it using an sret pointer.

Reply:
I believe the intent was to test a case where i128 gets returned in an LLVM IR struct. If that situation no longer occurs after that PR, then testing the sret case is fine.